pkg/agent/controller/endpoint_slice.go

```diff
@@ -136,3 +154,70 @@ func (c *EndpointSliceController) onLocalEndpointSliceSynced(obj runtime.Object,
 
 	return err != nil
 }
+
+func (c *EndpointSliceController) checkForConflicts(key, name, namespace string) (bool, error) {
+	epsList, err := c.syncer.GetLocalClient().Resource(endpointSliceGVR).Namespace(namespace).List(context.Background(), metav1.ListOptions{
+		LabelSelector: k8slabels.SelectorFromSet(map[string]string{
+			discovery.LabelManagedBy: constants.LabelValueManagedBy,
+			mcsv1a1.LabelServiceName: name,
+		}).String(),
+	})
+	if err != nil {
+		return true, errors.Wrapf(err, "error during conflict check for %q", key)
+	}
+
+	var prevServicePorts []mcsv1a1.ServicePort
+	var intersectedServicePorts []mcsv1a1.ServicePort
+	clusterNames := make([]string, 0, len(epsList.Items))
+	conflict := false
+
+	for i := range epsList.Items {
+		eps := c.serviceExportClient.toEndpointSlice(&epsList.Items[i])
+
+		servicePorts := c.serviceExportClient.toServicePorts(eps.Ports)
+		if prevServicePorts == nil {
+			prevServicePorts = servicePorts
+			intersectedServicePorts = servicePorts
+		} else if !slices.Equivalent(prevServicePorts, servicePorts, servicePortKey) {
```

Comment:
I believe this is based on assumption that service ports of aready exported service are not going to change, and we are only making sure newly exported service doesn't conflict.

What if someone changes service definition of alread exported service in a cluster?

Reply:
My quetsion was specific for scenario where service changes to introduce a conflict, but from workflow looks like it is covered too. Do we have an E2E for this scenario, conflict created after export creation?

Reply:
There's unit tests that cover the scenario but not E2E. We can add E2E later.